File: SimpleImageViewer/SimpleImageViewerNet/Album.cs
using System;
using System.Drawing;
using SpaceVIL.Common;
using SpaceVIL.Decorations;
using SpaceVIL.Core;
using SpaceVIL;
using System.IO;

namespace SimpleImageViewer
{
    public class Album : Prototype
    {
        public EventCommonMethodState OnDoubleClick;

        private HorizontalStack _topLayout = new HorizontalStack();
        internal ButtonToggle _expand = new ButtonToggle();
        private ImageItem _arrow = new ImageItem(DefaultsService.GetDefaultImage(EmbeddedImage.ArrowLeft, EmbeddedImageSize.Size32x32));
        public Label Name = new Label();
        private ButtonCore _remove = new ButtonCore();

        private HorizontalStack _bottomLayout = new HorizontalStack();
        private Label _pathLabel = new Label("Path:");
        private TextEdit _pathEdit = new TextEdit();
        private ButtonCore _pathBrowse = new ButtonCore();

        public String GetPath()
        {
            return _pathEdit.GetText();
        }

        public Album(String name, String path)
        {
            SetSizePolicy(SizePolicy.Expand, SizePolicy.Fixed);
            SetHeight(30);
            SetSpacing(0, 10);
            SetMargin(3, 5, 3, 5);
            SetAlignment(ItemAlignment.Left, ItemAlignment.Top);
            SetBackground(Color.Transparent);
            SetShadow(10, 0, 0, Color.FromArgb(200, 0, 0, 0));
            Name.SetText(name);
            _pathEdit.SetText(path);
            _bottomLayout.SetVisible(false);
        }

        public override void InitElements()
        {
            //top
            _topLayout.SetHeightPolicy(SizePolicy.Fixed);
            _topLayout.SetHeight(30);
            _topLayout.SetSpacing(5, 0);
            _topLayout.SetBackground(Color.FromArgb(20, 255, 255, 255));

            _expand.SetSize(20, 30);
            _expand.SetBackground(25, 25, 25);
            _expand.GetState(ItemStateType.Toggled).Background = Color.FromArgb(25, 25, 25);
            _expand.SetPadding(4, 9, 4, 9);
            _arrow.SetRotationAngle(180);
            _arrow.SetColorOverlay(Color.FromArgb(210, 210, 210));
            _arrow.KeepAspectRatio(true);

            Name.SetHeightPolicy(SizePolicy.Fixed);
            Name.SetHeight(30);
            Name.SetMargin(5, 0, 0, 0);
            Name.SetFontSize(16);
            Name.SetFontStyle(FontStyle.Bold);

            _remove.SetSize(12, 12);
            _remove.SetCustomFigure(new CustomFigure(false, GraphicsMathService.GetCross(12, 12, 3, 45)));
            _remove.SetBackground(100, 100, 100);
            _remove.SetAlignment(ItemAlignment.VCenter, ItemAlignment.Left);
            _remove.SetMargin(0, 0, 5, 0);

            //bottom
            _bottomLayout.SetHeightPolicy(SizePolicy.Fixed);
            _bottomLayout.SetHeight(30);
            _bottomLayout.SetSpacing(5, 0);
            _bottomLayout.SetAlignment(ItemAlignment.Left, ItemAlignment.Bottom);

            _pathLabel.SetWidthPolicy(SizePolicy.Fixed);
            _pathLabel.SetWidth(_pathLabel.GetTextWidth() + 5);
            // _pathLabel.SetMargin(5, 0, 0, 0);
            _pathLabel.SetFontSize(14);

            _pathBrowse.SetSize(30, 30);
            _pathBrowse.SetBackground(255, 255, 255, 20);
            _pathBrowse.SetPadding(7, 7, 7, 7);
            _pathBrowse.GetState(ItemStateType.Hovered).Background = Color.FromArgb(150, 255, 255, 255);

            AddItems(
                _topLayout,
                _bottomLayout
            );

            _topLayout.AddItems(
                _expand,
                Name,
                _remove
            );

            _bottomLayout.AddItems(
                _pathLabel,
                _pathEdit,
                _pathBrowse
            );

            _expand.AddItem(_arrow);
            _pathBrowse.AddItem(new ImageItem(DefaultsService.GetDefaultImage(EmbeddedImage.Pencil, EmbeddedImageSize.Size32x32), false));

            //events
            _expand.EventToggle += (sender, args) =>
            {
                _bottomLayout.SetVisible(!_bottomLayout.IsVisible());
                if (_bottomLayout.IsVisible())
                {
                    SetHeight(70);
                    _arrow.SetRotationAngle(90);
                }
                else
                {
                    SetHeight(30);
                    _arrow.SetRotationAngle(180);
                }
            };

            Name.EventMouseDoubleClick += (sender, args) =>
            {
                OnDoubleClick.Invoke(this);
            };

            _remove.EventMouseClick += (sender, args) =>
            {
                Remove();
            };
        }

        public void Remove()
        {
            GetParent().RemoveItem(this);
        }
    }
}

File: SimpleImageViewer/SimpleImageViewerJVM/src/main/java/SimpleImageViewerJVM/Album.java
package SimpleImageViewerJVM;

import java.awt.Color;
import java.awt.Font;

import com.spvessel.spacevil.*;
import com.spvessel.spacevil.Common.*;
import com.spvessel.spacevil.Core.*;
import com.spvessel.spacevil.Decorations.Figure;
import com.spvessel.spacevil.Decorations.ItemState;
import com.spvessel.spacevil.Flags.*;

public class Album extends VerticalStack {
    public EventCommonMethodState onDoubleClick = new EventCommonMethodState();

    private HorizontalStack _topLayout = new HorizontalStack();
    ButtonToggle _expand = new ButtonToggle();
    private ImageItem _arrow = new ImageItem(
            DefaultsService.getDefaultImage(EmbeddedImage.ARROW_LEFT, EmbeddedImageSize.SIZE_32X32));
    public Label name = new Label();
    private ButtonCore _remove = new ButtonCore();

    private HorizontalStack _bottomLayout = new HorizontalStack();
    private Label _pathLabel = new Label("Path:");
    private TextEdit _pathEdit = new TextEdit();
    private ButtonCore _pathBrowse = new ButtonCore();

    public String getPath() {
        return _pathEdit.getText();
    }

    public Album(String name, String path) {
        setSizePolicy(SizePolicy.EXPAND, SizePolicy.FIXED);
        setHeight(30);
        setSpacing(0, 10);
        setMargin(3, 5, 3, 5);
        setAlignment(ItemAlignment.LEFT, ItemAlignment.TOP);
        setBackground(new Color(0, 0, 0, 0));
        setShadow(10, 0, 0, new Color(0, 0, 0, 200));
        this.name.setText(name);
        _pathEdit.setText(path);
        _bottomLayout.setVisible(false);
    }

    @Override
    public void initElements() {
        // top
        _topLayout.setHeightPolicy(SizePolicy.FIXED);
        _topLayout.setHeight(30);
        _topLayout.setSpacing(5, 0);
        _topLayout.setBackground(new Color(255, 255, 255, 20));

        _expand.setSize(20, 30);
        _expand.setBackground(25, 25, 25);
        _expand.addItemState(ItemStateType.TOGGLED, new ItemState(new Color(25, 25, 25)));
        _expand.setPadding(4, 9, 4, 9);
        _arrow.setRotationAngle(180);
        _arrow.setColorOverlay(new Color(210, 210, 210));
        _arrow.keepAspectRatio(true);

        name.setHeightPolicy(SizePolicy.FIXED);
        name.setHeight(30);
        name.setMargin(5, 0, 0, 0);
        name.setFontSize(16);
        name.setFontStyle(Font.BOLD);
        name.setCursor(EmbeddedCursor.HAND);

        _remove.setSize(12, 12);
        _remove.setCustomFigure(new Figure(false, GraphicsMathService.getCross(12, 12, 3, 45)));
        _remove.setBackground(100, 100, 100);
        _remove.setAlignment(ItemAlignment.VCENTER, ItemAlignment.LEFT);
        _remove.setMargin(0, 0, 5, 0);

        // bottom
        _bottomLayout.setHeightPolicy(SizePolicy.FIXED);
        _bottomLayout.setHeight(30);
        _bottomLayout.setSpacing(5, 0);
        _bottomLayout.setAlignment(ItemAlignment.LEFT, ItemAlignment.BOTTOM);

        _pathLabel.setWidthPolicy(SizePolicy.FIXED);
        _pathLabel.setWidth(_pathLabel.getTextWidth() + 5);
        _pathLabel.setFontSize(14);

        _pathBrowse.setSize(30, 30);
        _pathBrowse.setBackground(255, 255, 255, 20);
        _pathBrowse.setPadding(7, 7, 7, 7);
        _pathBrowse.addItemState(ItemStateType.HOVERED, new ItemState(new Color(255, 255, 255, 150)));

        _pathEdit.setBackground(Color.white);

        addItems(_topLayout, _bottomLayout);

        _topLayout.addItems(_expand, name, _remove);

        _bottomLayout.addItems(_pathLabel, _pathEdit, _pathBrowse);

        _expand.addItem(_arrow);
        _pathBrowse.addItem(new ImageItem(
                DefaultsService.getDefaultImage(EmbeddedImage.PENCIL, EmbeddedImageSize.SIZE_32X32), false));

        // events
        _expand.eventToggle.add((sender, args) -> {
            _bottomLayout.setVisible(!_bottomLayout.isVisible());
            if (_bottomLayout.isVisible()) {
                setHeight(70);
                _arrow.setRotationAngle(90);

            } else {
                setHeight(30);
                _arrow.setRotationAngle(180);
            }
        });

        name.eventMouseDoubleClick.add((sender, args) -> {
            onDoubleClick.execute(this);
        });

        _remove.eventMouseClick.add((sender, args) -> {
            remove();
        });
    }

    public void remove()
    {
        getParent().removeItem(this);
    }
}